src/data/list/sort.lean

```diff
@@ -82,6 +82,22 @@ begin
   { exact list.pairwise_iff_nth_le.1 h a b hb H }
 end
 
+lemma sorted.rel_of_mem_take_of_mem_drop {l : list α} (h : list.sorted r l)
+  (k : ℕ) {x y : α} (hx : x ∈ list.take k l) (hy : y ∈ list.drop k l) :
+  r x y :=
+begin
```

Comment:
This seems very similar to `nth_le_of_sorted_of_le`. But that has some `is_refl` constraint that might be avoidable.

Reply:
```
lemma nth_le_of_sorted_of_lt {l : list α}
  (h : l.sorted r) {a b : ℕ} {ha : a < l.length} {hb : b < l.length} (hab : a < b) :
  r (l.nth_le a ha) (l.nth_le b hb) :=
list.pairwise_iff_nth_le.1 h a b hb hab
```
might be what you're looking for - it's currently missing. I think Eric's proof as it is still belongs in mathlib though

Reply:
```
lemma nth_le_of_sorted_of_lt {l : list α}
  (h : l.sorted r) {a b : ℕ} {ha : a < l.length} {hb : b < l.length} (hab : a < b) :
  r (l.nth_le a ha) (l.nth_le b hb) :=
list.pairwise_iff_nth_le.1 h a b hb hab

lemma sorted.rel_of_mem_take_of_mem_drop {l : list α} (h : list.sorted r l)
  (k : ℕ) {x y : α} (hx : x ∈ list.take k l) (hy : y ∈ list.drop k l) :
  r x y :=
begin
  obtain ⟨iy, hiy, rfl⟩ := nth_le_of_mem hy,
  obtain ⟨ix, hix, rfl⟩ := nth_le_of_mem hx,
  rw [nth_le_take', nth_le_drop'],
  apply nth_le_of_sorted_of_lt h,
  apply nat.lt_add_right,
  simp at hix,
  exact hix.left
end
```

Reply:
Can of course be golfed.

Reply:
Thanks, done and mildly golfed